### PR TITLE
Restrict milestone log parsing to cdo-logs/hosts prefix

### PR DESCRIPTION
**Pull request by wjordan**
This PR contains two incremental improvements to the milestone log counter script:
- Restrict `count-lines-of-code-from-milestone-logs` to only fetch logs from the `hosts` prefix, rather than all of the logs in the `cdo-logs` bucket.
- Rewrite the log parser using the official AWS SDK for Ruby client. Uses etags and md5 digests to only download+count modified *.log files, and only download+count the remainder of un-rotated logs whose initial content hasn't changed.

Manual tests on my local workstation for a fully-cached run (no changes to log files since the previous run):
- Original script: 742.5 seconds
- Modified script with `prefix: 'hosts'` change: 38.5 seconds
- v2 script: 3.5 seconds

The plan is to silently run the v2 script alongside the modified original script for several days, in order to ensure no regressions in the new version before switching to the new script for our live counter.

### CODE REVIEW

**Comment by wjordan** [comment]
Merging now to get the `prefix:'hosts'` fix in place and to allow side-by-side testing of the v2 parser. Will look for full code-review before switching to the v2 parser in the production CI script.

File: lib/cdo/analytics/milestone_parser.rb [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        end
+      end
+    else
+      # Assume cached .gz files without etag haven't been modified
+      return {'count' => cached, 'etag' => etag} if ext == '.gz' && cached.is_a?(Numeric)
+    end
+
+    debug "Downloading #{key}, #{fetch_bytes} bytes.."
+    log.get fetch_params
+    stub_fetch(key, path, fetch_bytes)
```
Comment (laurelfan) [inline lib/cdo/analytics/milestone_parser.rb]: is this just for testing?

File: lib/cdo/analytics/milestone_parser.rb [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    debug "Processing #{key}.."
+    # Sum up all values in the 10th tab-delimited column
+    count = partial_count + `#{ext == '.gz' ? 'gunzip -c' : 'cat'} #{path} | cut -f10 | awk '{s+=$1} END {print s}'`.to_i
```
Comment (laurelfan) [inline lib/cdo/analytics/milestone_parser.rb]: `zcat -f` will do that without having to check the file extension

File: lib/cdo/analytics/milestone_parser.rb (hunk 1 above)
Comment (wjordan) [inline lib/cdo/analytics/milestone_parser.rb]: yeah, just a testing hook, not ideal- probably cleaner to call a method that actually does something useful, and then override that in the test class

File: lib/cdo/analytics/milestone_parser.rb (hunk 2 above)
Comment (wjordan) [inline lib/cdo/analytics/milestone_parser.rb]: cool! :+1:

**Pull request merged**